libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/builder/NavRouteOptionsBuilder.kt

```diff
@@ -0,0 +1,369 @@
+package com.mapbox.navigation.core.routeoptions.builder
+
+import android.content.Context
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.Bearing
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.base.extensions.applyLanguageAndVoiceUnitOptions
+
+@ExperimentalPreviewMapboxNavigationAPI
+interface NoWaypointsOptionsBuilder {
+    fun fromStartLocation(
+        point: Point,
+        bearing: Double? = null,
+        zLevel: Int?
+    ): WaypointsInProgressBuilder
```

Comment:
We cannot define interfaces with default arguments because we won't be able to extend the parameter set for Java consumers. Adding a parameter to an interface function will always be a breaking change since `@JvmOverloads` is not available in interfaces.

Unfortunately it looks like we'll need to split `NavRouteOptionsBuilder` into 3 separate classes that pass the data between each other in a chain, unless you have other ideas?

Reply:
If we can do that, it might actually make the experience more consistent with the rest of the build methods that do use functional extensions as arguments.